### PR TITLE
Use CRUD for database operations

### PR DESCRIPTION
**Pull request by moonso**
This PR changes so that the database operations are moved to CreatReadUpdataDelete directory

### CODE REVIEW

File: NIPTool/adapter/plugin.py [hunk 1]
```diff
@@ -12,63 +14,10 @@ def setup(self, db_name: str):
...
-        self.db = self.client[db_name]
-        self.db_name = db_name
-        self.sample_collection = self.db.sample
-        self.batch_collection = self.db.batch
-        self.user_collection = self.db.user
+        self.db: Database = self.client[db_name]
+        self.db_name: str = db_name
+        self.sample_collection: Collection = self.db.sample
+        self.batch_collection: Collection = self.db.batch
+        self.user_collection: Collection = self.db.user
```
Comment (mayabrandi) [inline NIPTool/adapter/plugin.py:21]: I like this

File: NIPTool/crud/find.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    return [batch_to_model(raw_batch) for raw_batch in raw_batches]
+
+
+def sample_aggregate(adapter: NiptAdapter, pipe: list) -> List[Sample]:
+    """Aggregates a query pipeline on the sample collection"""
+    raw_samples: Iterable[dict] = adapter.sample_collection.aggregate(pipe)
+    return [sample_to_model(raw_sample) for raw_sample in raw_samples]
+
+
+def batch_aggregate(adapter: NiptAdapter, pipe: list) -> List[Batch]:
```
Comment (mayabrandi) [inline NIPTool/crud/find.py:48]:
```suggestion
def batch_aggregate(adapter: NiptAdapter, pipe: list) -> list:
```

File: NIPTool/crud/find.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    return batch_to_model(raw_batch)
+
+
+def batches(adapter: NiptAdapter) -> List[Batch]:
+    """Return all batches from the batch collection"""
+    raw_batches: Iterable[dict] = adapter.batch_collection.find()
+    return [batch_to_model(raw_batch) for raw_batch in raw_batches]
+
+
+def sample_aggregate(adapter: NiptAdapter, pipe: list) -> List[Sample]:
```
Comment (mayabrandi) [inline NIPTool/crud/find.py:42]:
```suggestion
def sample_aggregate(adapter: NiptAdapter, pipe: list) -> list:
```

File: NIPTool/crud/find.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    raw_batch: dict = adapter.batch_collection.find_one({"_id": batch_id})
+    return batch_to_model(raw_batch)
+
+
+def batches(adapter: NiptAdapter) -> List[Batch]:
+    """Return all batches from the batch collection"""
+    raw_batches: Iterable[dict] = adapter.batch_collection.find()
+    return [batch_to_model(raw_batch) for raw_batch in raw_batches]
+
+
```
Comment (mayabrandi) [inline NIPTool/crud/find.py:41]: Aggragations do not return lists of Sample or Batch objects. The format of the content is totally depending on the pipe that is inserted

File: NIPTool/crud/find.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
+from typing import Iterable, List
+
+from NIPTool.adapter import NiptAdapter
+from NIPTool.models.database import Batch, User, Sample
+
+
```
Comment (mayabrandi) [inline NIPTool/crud/find.py:6]: supernice with a crude/find

File: NIPTool/models/database/sample.py [hunk 6]
```diff
@@ -76,3 +76,7 @@ class Sample(BaseModel):
     UnfilteredCNVcalls: Optional[int]
     CNVSegment: Optional[str]
     segmental_calls: Optional[str]
+    comment: Optional[str]
+    include: Optional[bool]
+    change_include_date: Optional[str]
+    status_change_abnormality: Optional[str]
```
Comment (mayabrandi) [inline NIPTool/models/database/sample.py:82]:
```suggestion
    status_change_abnormality: Optional[str]
        status_13: Optional[str]
    status_18: Optional[str]
    status_21: Optional[str]
    status_X0: Optional[str]
    status_XXX: Optional[str]
    status_XXY: Optional[str]
    status_XYY: Optional[str]
    status_change_T13: Optional[str]
    status_change_T18: Optional[str]
    status_change_T21: Optional[str]
    status_change_TX0: Optional[str]
    status_change_TXXX: Optional[str]
    status_change_TXXY: Optional[str]
    status_change_TXYY: Optional[str]
```

Comment (mayabrandi) [reply NIPTool/models/database/sample.py:82]: These are needed as well

File: NIPTool/models/database/batch.py [hunk 7]
```diff
@@ -4,7 +4,7 @@
 
 
 class Batch(BaseModel):
-    SampleProject: str
+    batch_id: str
```
Comment (mayabrandi) [inline NIPTool/models/database/batch.py:7]: There are issues with this. I still think we need to have a separate model for the fluffy file unfortunately. But maybe we can find a way to work around it.

File: NIPTool/server/web_app/api/api_v1/endpoints/update.py [hunk 8]
```diff
@@ -1,66 +1,79 @@
...
+from NIPTool.crud import find, update
+from NIPTool.models.database import Sample, User
 from NIPTool.server.web_app.api.deps import get_nipt_adapter
-from datetime import datetime
+from NIPTool.server.web_app.utils import *
 
 router = APIRouter()
 
+LOG = logging.getLogger(__name__)
+
```
Comment (mayabrandi) [inline NIPTool/server/web_app/api/api_v1/endpoints/update.py:17]: This whole function will be re written and split into smaller. But lets do it later

**Review by mayabrandi — COMMENTED** [review]
very nice input Måns! I like the idea of crud/update and crud/insert. Didnt have time to test them now though. I want to test before merging.

File: NIPTool/adapter/plugin.py (hunk 1 above)
Comment (moonso) [reply NIPTool/adapter/plugin.py:21]: :)

File: NIPTool/crud/find.py (hunk 4 above)
Comment (moonso) [reply NIPTool/crud/find.py:41]: Yes I suspected that. I suggest that we implement functions for the different types of aggregations. Do you think that would be possible ?

File: NIPTool/models/database/sample.py (hunk 6 above)
Comment (moonso) [reply NIPTool/models/database/sample.py:82]: can we talk about this on Monday?

File: NIPTool/models/database/batch.py (hunk 7 above)
Comment (moonso) [reply NIPTool/models/database/batch.py]: Yes I hope so

File: NIPTool/server/web_app/api/api_v1/endpoints/update.py (hunk 8 above)
Comment (moonso) [reply NIPTool/server/web_app/api/api_v1/endpoints/update.py:17]: Sound good

**Pull request closed without merging**